### PR TITLE
Add DashScope API based multimodal service functions in AgentScope library

### PR DESCRIPTION
**Pull request by PengHongyiNTU**
This PR introduces three new services to the service toolkit: dashscope_text_to_image, dashscope_image_to_text, and dashscope_text_to_audio.

also fix a small error in examples/conversation_with_customized_services

### CODE REVIEW

**Comment by garyzhang99** [comment]
The title of this PR isn't quite accurate.

**Comment by garyzhang99** [comment]
Although it might be a bit of a hassle, please consider adding corresponding tests for the new added service functions. 
You may refer to the examples in `tests/execute_python_code_test.py` and `tests/dashscope_test.py`, utilizing mocks to simulate the return values of the relevant `dashscope` libraries. This way, if any bugs arise in the future due to changes in the library, we can quickly pinpoint them.

File: src/agentscope/service/multi_modality/dash_scope.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        print(image_to_text(image_url, query_prompt)) gives:
+        {'status': 'SUCCESS', 'content': 'A beautiful sunset in the mountains'}
+    """
+    dashscope.api_key = api_key
+    if not isinstance(image_urls, tuple):
+        image_urls = (image_urls,)
+    contents = []
+    for image_url in image_urls:
+        # check image url is local content or remote content
+        if not image_url.startswith(("http://", "https://")):
```
Comment (DavdGao) [inline src/agentscope/service/multi_modality/dash_scope.py:126]: Suggested to reuse the DashScopeMultimodalModelWrapper in AgentScope, which already handles the local/web url and prompt construct

File: src/agentscope/service/multi_modality/dash_scope.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        prompt = "A beautiful sunset in the mountains"
+        print( dashscope_text_to_image(prompt)) gives:
+        {'status': 'SUCCESS',
+        'content': {'fig_paths':
+            ['FIG_RELATEIVE_PATH1',
+            'FIG_RELATEIVE_PATH2']}}
+    """
+
+    dashscope.api_key = api_key
+    res = dashscope.ImageSynthesis.call(
```
Comment (DavdGao) [inline src/agentscope/service/multi_modality/dash_scope.py:57]: Suggested to reuse the model wrapper in AgentScope

File: src/agentscope/service/multi_modality/dash_scope.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        err_msg += res.message
+        return ServiceResponse(
+            ServiceExecStatus.ERROR,
+            {"error": err_msg},
+        )
+
+
+def dashscope_image_to_text(
+    image_urls: Union[str, Tuple[str, ...]],
+    prompt: str,
```
Comment (DavdGao) [inline src/agentscope/service/multi_modality/dash_scope.py:98]: The docstring misses the parameter `prompt`, and should we consider to provided a default prompt here? e.g. describe the figure.

**Review by DavdGao — COMMENTED** [review]
Please see inline comments.

File: src/agentscope/service/multi_modality/dash_scope.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    api_key: str,
+    model: str = "qwen-vl-plus",
+) -> ServiceResponse:
+    """Generate text based on an image.
+
+    Args:
+        image_urls (`str`): the url of single or multiple images.
```
Comment (xieyxclack) [inline src/agentscope/service/multi_modality/dash_scope.py:105]: I am not sure, whether `image_urls` can contain both local paths and remote urls?

**Review by xieyxclack — COMMENTED** [review]
LGTM. 
BTM, it may be not convenient for users to have to pass the same api_key every time they use dashscope_xx_to_xx. Is it possible for users to set up an api_key with export api_key=xxx instead? @DavdGao

File: src/agentscope/service/multi_modality/dash_scope.py (hunk 4 above)
Comment (DavdGao) [reply src/agentscope/service/multi_modality/dash_scope.py:105]: The local url requires special prefix in the url

File: src/agentscope/service/multi_modality/dash_scope.py (hunk 1 above)
Comment (PengHongyiNTU) [reply src/agentscope/service/multi_modality/dash_scope.py:126]: The DashScopeMultimodalModelWrapper handles it through a protected member function called _convert_url. Therefore, I choose to stay with the original version.

Comment (PengHongyiNTU) [reply src/agentscope/service/multi_modality/dash_scope.py:126]: Can also use Wrapper.format but it require the input to be a MessageBase. Seems a bit heavy for me.

**Review by PengHongyiNTU — COMMENTED** [review]
Need modify test accordingly

**Review by DavdGao — APPROVED** [review]
LGTM

**Pull request merged**